components/ui/WeekdaySelect.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from "react";
+
+export const WeekdaySelect = (props) => {
+  const [activeDays, setActiveDays] = useState(
+    [...Array(7).keys()].map((v, i) => (props.defaultValue || []).includes(i))
+  );
+
+  const days = ["S", "M", "T", "W", "T", "F", "S"];
+
+  useEffect(() => {
+    props.onSelect(activeDays.map((v, idx) => (v ? idx : -1)).filter((v) => v !== -1));
+  }, [activeDays]);
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'props'. Either include it or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useEffect call and refer to those specific props inside useEffect.](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689306)
```suggestion
  }, [activeDays, props]);
```